### DISCUSSION TITLE
Increase user-friendliness for "regular" users

### DISCUSSION THREAD

**Discussion by serkor1** [posit-dev/positron#5331] [Feedback]
Hi,

I find that Positron is lacking a user-friendly aspect, if I can put it like that. This is mainly due to the fact that "stuff" is missing under the hood which RStudio brings by default - like, for example, MAKE and Pandoc support out of the box.

--------

I, as a package developer, have developed a humble and insignificant skillset,  which enables me to fix some of the challenges when missing basic stuff like paths to various binaries that makes it all work under the hood. Recently, I posted a discussion on MAKE-files which was an easy fix (see discussion https://github.com/posit-dev/positron/discussions/5326). But why not make it work by default?

It turns out that (for me at least) I still needed to install Pandoc from the official github repository to fix the following error:

```bash
pandoc is required to build R markdown vignettes
```

This has been discussed in Issue https://github.com/posit-dev/positron/issues/1466, and I believe it has been attempted to be fixed in PR https://github.com/posit-dev/positron/pull/2547 - but it seems this is not the case.

Both MAKE and Pandoc is essential parts of the development workflow of R packages - and although you may argue that the developer label should imply a higher baseline skillset, it is my understanding that Positron is also targeted regular users (who may or may not know other languages than R). I hope that you will bring many of the essential features that we need with R development in general at installation in the same way that RStudio does (Which I just realized I completely took for granted).

Best,

**Comment by juliasilge** [comment]
We definitely believe that pandoc is working for most users and it's unexpected that you would need to install pandoc separately from what is bundled in Positron. It would be useful for us to have more info on the specific task you were trying to approach and how you saw this error message.

We plan that eventually Positron will be a good fit for even beginners, but currently it is a beta product with some rough edges. You might check out ["Is Positron for Me?"](https://positron.posit.co/start.html#is-positron-for-me) to see who we think it is currently a good fit for.

**Reply by serkor1** [reply]
Hi @juliasilge 

What information do you need? 😊I was running R CMD Build via the built-in terminal.

The package vignettes would not build and give the missing Pandoc message.

I'll give a more reproducible example when I know what you need!

Best,

**Reply by serkor1** [reply]
Also, it was not a critique nor did I want to imply that it doesn't work for most users - it is, in fact, working like a charm for me in my everyday work, and I love everything about it. I am aware of the premise that its still in beta and therefore will have rough edges.

**Comment by jennybc** [comment]
> What information do you need? 😊I was running R CMD Build via the built-in terminal.

The Pandoc that ships inside Positron should be available if you're using any Positron-provided gestures, such as the command "R: Build R Package", also accessible via Ctrl/Cmd + Shift + B (at least, with default keybindings). We explicitly inject the path to the built-in Pandoc as the `RSTUDIO_PANDOC` environment variable when we run package dev tasks (or render `.Rmd`) in a terminal.

In an arbitrary terminal running inside Positron that you've launched, this won't happen and it's not clear that we even should do that. If that's how you prefer to run `R CMD build`, you'll need to take the same measures to make Pandoc discoverable that you'd need to take outside of Positron. Such as ensuring Pandoc can be found on your `PATH` or explicitly advertising its location via the `RSTUDIO_PANDOC` env var.

In my past RStudio life, I have also occasionally wanted to do, e.g., `R CMD check` from a terminal, which suffers from the exact same problem and sometimes I would do this as a quick one-off solution:

```
export PATH="/Applications/RStudio.app/Contents/MacOS/pandoc/:$PATH"
R CMD check --no-manual --as-cran gargle_1.1.0.9000.tar.gz 
```

**Reply by serkor1** [reply]
Thank you for the thorough answer - I am aware of the "R: Build R Package" option. However, this option uses {devtools} which in my current package under development is suboptimal; so I need the `R CMD check` and `R CMD build`. And running these commands inside an arbitrary terminal using RStudio is without any issues, at all. In fact, this is where I started running these commands before I moved to Positron.

The reason that one might need to use the terminal commands instead of the {devtools} approach is that, for packages written with Rcpp, {devtools} overwrites the RcppExports.R-file. See for example this [SO](https://stackoverflow.com/questions/57437298/avoid-regenerating-rcppexports-when-using-devtoolstest) or this issue https://github.com/r-lib/devtools/issues/1248. 

I can't comment on whether it is good or bad practice to use the terminal over, say, the built-in functionality with {devtools} - I am just an Economist, and not a 'real' developer. But I do believe that Positron should implement the same baseline features as RStudio; in this particular outlier case, too! 😄

**Comment by juliasilge** [comment]
I opened https://github.com/posit-dev/positron/issues/5357 to discuss (as well as track/prioritize) the possibility of making the bundled pandoc available on the path in terminals.

**Reply by serkor1** [reply]
Thank you for this. 😊
